### PR TITLE
Update common models

### LINKED ISSUE #288: Add JobName
Define the type `JobName` to encapsulate a job name

### LINKED ISSUE #289: Add OwnerName
Define the type `OwnerName` to encapsulate a owner name

### LINKED ISSUE #290: Add tests for DbTableVersion
Let's add test coverage for `DbTableVersion`

### LINKED ISSUE #291: Add support for underscores in URNs
Let's add support for underscores (`_`) in dataset URNs

### LINKED ISSUE #292: Define Junit category UnitTests
Let's define the Junit category `UnitTests` to run unit tests only

see https://github.com/junit-team/junit4/wiki/categories

### LINKED ISSUE #293: Define gradle task to run unit tests only
Let's define a gradle task to run unit tests only leveraging the category define in #292

### LINKED ISSUE #294: Rename common models that wrap names
For clarity, let's rename the following types #166 :

```java
Dataset.java   => DatasetName.java
Db.java        => DbName.java
DbSchema.java  => DbSchemaName.java
DbTable.java   => DbTableName.java
Namespace.java => NamespaceName.java 
```

### PR DESCRIPTION
**Pull request by wslulciuc**
This PR makes the following additions/changes to our common models:

- Types that wrap names now end with `*Name`: for example, `Db` wraps a database name and has been renamed to `DbName` (closes #294)
- The factory method `fromString()` annotated with `@JsonCreator` replaces `of()` to ensure types can be deserialized correctly
- Add clearer descriptions to tests
- Define the junit category `UnitTests` (closes #292)
- Define gradle task to run unit tests only (closes #293)
- Add support for underscores (`_`) in URNs (closes #291)
- Add tests for **class** [DbTableVersion](https://github.com/MarquezProject/marquez/blob/master/src/main/java/marquez/service/models/DbTableVersion.java) (closes #290)
- Add **class** [JobName](https://github.com/MarquezProject/marquez/blob/update-common-models/src/main/java/marquez/common/models/JobName.java) (closes #288)
- Add **class** [OwnerName](https://github.com/MarquezProject/marquez/blob/update-common-models/src/main/java/marquez/common/models/OwnerName.java) (closes #289)

### Running tests by category

```
# All tests
$ ./gradlew test

# Unit tests only
$ ./gradlew testUnit
```

**NOTE:** This PR contains multiple changes that, though are related, should have been introduced in separate PRs and will not be the norm going forward :)

### Resources

- [Junit Categories](https://github.com/junit-team/junit4/wiki/categories)

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -40,6 +40,12 @@ compileTestJava {
   options.incremental = true
 }
 
+task testUnit(type: Test) {  
+  useJUnit {
+    includeCategories 'marquez.UnitTests'
+  }
+}
+
```
Comment (ashulmanWeWork) [inline build.gradle]: This should probably go into a different review just b/c it's purpose is different than the rest of the changes.

File: src/main/java/marquez/common/models/DatasetUrn.java [hunk 2]
```diff
@@ -10,38 +10,39 @@
...
-
-  public static DatasetUrn of(String value) {
-    return new DatasetUrn(value);
-  }
-
   private DatasetUrn(@NonNull final String value) {
     if (!URN_PATTERN.matcher(value).matches()) {
       throw new IllegalArgumentException(
-          "A urn must contain only letters (a-z, A-Z), numbers (0-9), and "
+          "A urn must contain only letters (a-z, A-Z), numbers (0-9), or underscores (_) and "
```
Comment (ashulmanWeWork) [inline src/main/java/marquez/common/models/DatasetUrn.java]: There are a lot of test changes so very possible I missed it, but is there a unit test for this?

File: build.gradle (hunk 1 above)
Comment (wslulciuc) [reply build.gradle]: We use the `UnitTests` category below in `ConnectionUrlTest`, so it's (loosely) relevant to the change, but should have been introduced in a separate PR

**Comment by wslulciuc** [comment]
@sshah-wework : Yeah, there's a lot going on, and I'm totally not following our own guidelines on PRs:

> keep diffs small and self-contained

Given that the changes are more around naming + aesthetics, I say we're ok to merge, but going forward, we'll have to block the PR :)

File: src/main/java/marquez/common/models/DatasetUrn.java (hunk 2 above)
Comment (wslulciuc) [reply src/main/java/marquez/common/models/DatasetUrn.java]: Actually, I don't have an explicit test for this case! Opened an issue : https://github.com/MarquezProject/marquez/issues/312

**Pull request merged**